src/thread.cpp

```diff
@@ -0,0 +1,184 @@
+#include <time.h>
+#include "os.h"
+#include "thread.h"
+
+pthread_key_t ProfiledThread::_tls_key;
+int ProfiledThread::_buffer_size = 0;
+std::atomic<int> ProfiledThread::_running_buffer_pos(0);
+std::vector<ProfiledThread*> ProfiledThread::_buffer(0);
```

Comment:
nit: not need to initialize `std::vector`, default constructor will construct an empty vector:
```suggestion
std::vector<ProfiledThread*> ProfiledThread::_buffer;
```